targets/TARGET_Maxim/TARGET_MAX32670/TARGET_XDOT_MAX32670/SystemInit.c

```diff
@@ -0,0 +1,120 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of MultiTech nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "max32670.h"
+#include "gcr_regs.h"
+#include "mxc_sys.h"
+#include "pwrseq_regs.h"
+
+#define XDOT_ERFO_FREQ      24000000    // Change to 24000000 for xDot 1.5 Rev A
+
+void SystemCoreClockUpdateXdot(void)
+{
+    uint32_t base_freq, div, clk_src;
+
+    // Get the clock source and frequency
+    clk_src = (MXC_GCR->clkctrl & MXC_F_GCR_CLKCTRL_SYSCLK_SEL);
+    switch (clk_src)
+    {
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_EXTCLK:
+            base_freq = EXTCLK_FREQ;
+            break;
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_ERFO:
+            base_freq = XDOT_ERFO_FREQ;
+            break;
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_INRO:
+            base_freq = INRO_FREQ;
+            break;
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_IPO:
+        base_freq = IPO_FREQ;
+            break;
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_IBRO:
+        base_freq = IBRO_FREQ;
+            break;
+        case MXC_S_GCR_CLKCTRL_SYSCLK_SEL_ERTCO:
+            base_freq = ERTCO_FREQ;
+            break;
+        default:
+            // Codes 001 and 111 are reserved.
+            // This code should never execute, however, initialize to safe value.
+            base_freq = HIRC_FREQ;
+            break;
+    }
+    // Get the clock divider
+    if (clk_src == MXC_S_GCR_CLKCTRL_SYSCLK_SEL_IPO)
+    {
+        base_freq = base_freq >> ((MXC_GCR->clkctrl & MXC_F_GCR_CLKCTRL_IPO_DIV)>> MXC_F_GCR_CLKCTRL_IPO_DIV_POS);
+    }
+    div = (MXC_GCR->clkctrl & MXC_F_GCR_CLKCTRL_SYSCLK_DIV) >> MXC_F_GCR_CLKCTRL_SYSCLK_DIV_POS;
+
+    SystemCoreClock = base_freq >> div;
+
+    // base_freq1 = base_freq;
```

Comment:
I've missed this one , please remove dead code

Reply:
Dead code removed and pushed.
For test results, what type of logs are you looking for? We built our bootloader code and AT application code for our new LoRaWAN endpoint. We run this combination of code through and extensive series of automated and manual tests. That includes full coverage of our AT commands, LoRaWAN communication and certification, low power modes and firmware updates. This exercises all the onboard hardware of our modules as well as off board SPI and GPIOs.

Reply:
Most of the time we get logs from greentea test run (executing all the tests for a new target). To answer: how was this tested, are all tests green, any known issues?